internal/resources/azure/monitor_data_collection_rule.go

```diff
@@ -0,0 +1,75 @@
+package azure
+
+import (
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+// MonitorDataCollectionRule struct represents an Azure Monitor Data Collection Rule.
+//
+// Resource information: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_data_collection_rule
+// Pricing information: https://azure.microsoft.com/en-in/pricing/details/monitor/
+type MonitorDataCollectionRule struct {
+	Address string
+	Region  string
+
+	MonthlyCustomMetricsIngestGB *int64 `infracost_usage:"monthly_custom_metrics_samples"`
```

Comment:
Should this be samples?
```suggestion
	MonthlyCustomMetricsSamples *int64 `infracost_usage:"monthly_custom_metrics_samples"`
```